src/Plotting/StrengthEnvelope.jl

```diff
@@ -0,0 +1,319 @@
+        if F > 0
+            ϕ = Mat.Plasticity[1].ϕ.val
+            c = Mat.Plasticity[1].C.val
+            τ[i] = Pres * sind(ϕ) + cosd(ϕ) * c
+        end
+    end
+
+    return τ
+end
+
+function redimensionalize(x, nz, param_dim::Unitful.FreeUnits, CharDim)
+    x_val = zeros(Float64, nz)
+    x_dim = dimensionalize(x, param_dim, CharDim)
+    for i = 1 : nz
+        x_val[i] = x_dim[i].val
+    end
+    return x_val
+end
+
+"""
+    StrengthEnvelopeSliders(MatParam, Thickness, TempType=LinearTemp()::AbstractThermalStructure; nz::Int64)
+
+Creates a GUI that plots a 1D strength envelope. In the GUI, temperature profile and strain rate can be adjusted. The Drucker-Prager plasticity uses lithostatic pressure.
+
+Parameters:
+- MatParam:  a tuple of materials (including the following properties: Phase, Density, CreepLaws, Plasticity)
+- Thickness: a vector listing the thicknesses of the respective layers (should carry units)
+- TempType:  the type of temperature profile (LinearTemp=default, HalfspaceCoolingTemp, ConstantTemp)
+- nz:        optional argument controlling the number of points along the profile (default = 101)
+
+# Example:
+```julia-repl
+julia> using GLMakie
+julia> MatParam = (SetMaterialParams(Name="UC", Phase=1, Density=ConstantDensity(ρ=2700kg/m^3), CreepLaws = SetDislocationCreep("Wet Quartzite | Ueda et al. (2008)"), Plasticity = DruckerPrager(ϕ=30.0, C=10MPa)),
+                   SetMaterialParams(Name="MC", Phase=2, Density=Density=ConstantDensity(ρ=2900kg/m^3), CreepLaws = SetDislocationCreep("Plagioclase An75 | Ji and Zhao (1993)"), Plasticity = DruckerPrager(ϕ=20.0, C=10MPa)),
+                   SetMaterialParams(Name="LC", Phase=3, Density=PT_Density(ρ0=2900kg/m^3, α=3e-5/K, β=1e-10/Pa), CreepLaws = SetDislocationCreep("Maryland strong diabase | Mackwell et al. (1998)"), Plasticity = DruckerPrager(ϕ=30.0, C=10MPa)));
+julia> Thickness = [15,10,15]*km;
+
+julia> StrengthEnvelope(MatParam, Thickness, LinearTemp())
+```
+"""
+function StrengthEnvelope(MatParam::NTuple{N, AbstractMaterialParamsStruct}, Thickness::Vector{U}, TempType::AbstractThermalStructure=LinearTemp(), mode::String="normal"; nz::Int64=101) where {N, U}
+
+    # hardcoded input
+    g         = 9.81m/s^2
+
+    # nondimensionalize
+    CharDim   = GEO_units(length=10km, temperature=1000C, stress=10MPa, viscosity=1e20Pas)
+    MatParam  = nondimensionalize(MatParam, CharDim)
+    Thickness = nondimensionalize(Thickness, CharDim)
+    g         = nondimensionalize(9.81m/s^2, CharDim)
+    
+    # derived properties
+    nLayer    = length(MatParam)
+    Inter     = cumsum(Thickness)
+    dz        = Inter[end] / (nz-1)
+    z         = collect(0:dz:Inter[end])
+
+    # distribute phases
+    Phases    = ones(Int64, nz) * MatParam[1].Phase
+    for i = 1 : nLayer - 1
+        Phases[z .> Inter[i]] .= MatParam[i+1].Phase
+    end
+
+    # check temperature structure
+    if typeof(TempType) == LinearTemp
```

Comment:
Since you are checking for the type of `TempType`, you could remove all these branching by using multiple dispatch

Reply:
> Since you are checking for the type of `TempType`, you could remove all these branching by using multiple dispatch

I could but maybe the compute temperature functions will at some point be used by some other functions as well and I would like to keep them as general as possible. I agree that the current way is a bit long but I like that it is very easy to follow.

Reply:
you could make a separate computational routine that does all the work required here, and simply call this routine from the plotting routine. In that case you don't duplicate code and it is easier to maintain (plus can be tested)

Reply:
> you could make a separate computational routine that does all the work required here, and simply call this routine from the plotting routine. In that case you don't duplicate code and it is easier to maintain (plus can be tested)

So I ended up going down this road and stumbled upon a general issue which causes problems for the halfspace-cooling case:
Why does this work? 
julia> 500m + 1km
1500 m

and this does not?
julia> 500m / 1km
500.0 m km^-1.0

Reply:
Good point. These type of calculations use the [Unitful.jl](https://github.com/PainterQubits/Unitful.jl) package, and it seems that this package does not support that:

```julia
julia> using Unitful
julia> 1000u"km"
1000 km
julia> 1u"km" + 5u"m"
1005 m
julia> 1u"km" / 5u"m"
0.2 km m⁻¹
julia> 1u"km" * 5u"m"
5 m km
julia> 1u"km" - 5u"m"
995 m
```